packages/vue-docgen-api/src/script-setup-handlers/utils/tsUtils.ts

```diff
@@ -150,7 +150,10 @@ export function getTypeDefinitionFromIdentifier(
 		},
 		visitTSTypeAliasDeclaration(nodePath) {
 			if (bt.isIdentifier(nodePath.node.id) && nodePath.node.id.name === typeName) {
-				typeBody = nodePath.get('typeAnnotation', 'members')
+				const typeAnnotation = nodePath.get('typeAnnotation', 'members')
+				if(typeAnnotation.value) {
+					typeBody = typeAnnotation
+				}
```

Comment:
What would you think of actually resolving the value like this?

```suggestion
				const typeAnnotation = nodePath.get('typeAnnotation')
				if(bt.isTSTypeLiteral(typeAnnotation.node)) {
					typeBody = typeAnnotation.get('members')
				}else if(bt.isTSTypeReference(typeAnnotation.node)){
				        typeBody = getTypeDefinitionFromIdentifier(
								astPath,
								typeAnnotation.node.typeName.name,
								opt
							)
				}
```

Reply:
@elevatebart 
Thanks for the review!
Oh! You're right, recursion is a great way to resolve reference types.
I have incorporated your suggestion.

P.S
I would like to be able to retrieve comments about types that utilize Intersection types and utility types as well. 
eg) 
```ts
type Foo = {
  /**
   * This is name
   */
  name: string
  /**
   * This is age
   */
  age: number
}

type Bar = {
  /**
   * This is sex
   */
  sex: male | female
}

type Baz = Foo & Baz

defineProps<Bar>()
```

To achieve this, I have a feeling that we may need to leverage TypeScript's API, and it seems like it might have a somewhat challenging learning curve.

Reply:
Going to go down that route, I would rather have storybook start using [vue-component-meta](https://github.com/vuejs/language-tools/tree/master/packages/vue-component-meta) instead of vue-docgen-api. I have less and less time to maintain this lib as we go and [vue-component-meta](https://github.com/vuejs/language-tools/tree/master/packages/vue-component-meta) is becoming more and more standard. GIven that it's part of the official vue org, I would say it has more future than the one here.

I had a good run ;)

Reply:
I found this issue on the storybook side.
https://github.com/storybookjs/storybook/pull/22285

In any case, thanks for developing a great package!